lib/syntax.ml

```diff
@@ -10,12 +10,21 @@ let debug = ref false
 type env = {
   tables : Tables.table list;
   schema : table_name Schema.Source.t;
+  ctes : Tables.table list;
```

Comment:
ctes cannot be part of tables?

Reply:
I waited for this question. Yes, they can be tables. I didn't add some opaque type to differentiate them. Moreover, it reduces the count of code changes, but adding cte to the tables list `Tables.add cte` makes this cte living during the all lifecycle, not only for the current statement lifecycle. If you mean tables from env ,  they are intermediate entity for schema, for storing aliases, and what more important we don't use them as sources, currently we use globals table as sources:
```ocaml
  | `Table s ->
    let (name,s) = Tables.get ~env s in
    let sources = (name :: option_list alias) in
    let s3 = List.map (fun attr -> { Schema.Source.Attr.attr; sources }) s  in
    s3, [], List.map (fun name -> name, s) sources
```

This is why I decided to add this small module which merges ctes with global tables living during all statement handling 
```ocaml
module Tables_with_derived = struct
  open Tables

  let get ~env name = get_from  (env.ctes @ Tables.all()) name

  let get_from ~env name = get_from  (env.ctes @ env.tables) name
end
```

Reply:
> I waited for this question

then should have just put this comment preventively in code. Need to put now anyway :)

Reply:
Ready

Reply:
i don't see :thinking: (don't see the comment on Tables_with_derived)

Reply:
https://github.com/ahrefs/sqlgg/pull/25/files#diff-e5b2ac5b61488c46149361560003e3185b33725d73f243f2cce2fc3020757438R13-R18
I added under ctes, but okay I will add similiar to Tables_with_derived

Reply:
<img width="759" alt="screen_ctes" src="https://github.com/user-attachments/assets/5167f3e5-b4b6-4ba8-83e8-b642bed150ca">
I added it under Tables_with_derived, but it might be unnecessary since there's already a comment above.

Reply:
idk how I missed it, yes, it should be enough